### PR TITLE
Allow more access between Kali and Teamserver instances

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request allows full TCP and UDP access between Kali and Teamserver instances on ports 5000-5999.

## 💭 Motivation and context ##

Resolves cisagov/cool-system-internal#79.

## 🧪 Testing ##

Automated testing passes.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

File: kali_sg.tf [hunk 1]
```diff
@@ -65,6 +51,30 @@ resource "aws_security_group_rule" "ingress_from_anywhere_to_kali_via_allowed_po
   to_port           = each.value["to_port"]
 }
 
+# Allow unfettered access between Kali and Teamserver instances
```
Comment (dav3r) [inline kali_sg.tf]:
```suggestion
# Allow access between Kali and Teamserver instances on ports 5000-5999
# This port range was requested for use by assessment operators
```

File: teamserver_sg.tf [hunk 2]
```diff
@@ -81,3 +67,27 @@ resource "aws_security_group_rule" "ingress_from_anywhere_to_teamserver_via_allo
   from_port         = each.value["from_port"]
   to_port           = each.value["to_port"]
 }
+
+# Allow unfettered access between Teamserver and Kali instances
```
Comment (dav3r) [inline teamserver_sg.tf]:
```suggestion
# Allow access between Teamserver and Kali instances on ports 5000-5999
# This port range was requested for use by assessment operators
```

File: kali_sg.tf [hunk 3]
```diff
@@ -65,6 +51,30 @@ resource "aws_security_group_rule" "ingress_from_anywhere_to_kali_via_allowed_po
...
+resource "aws_security_group_rule" "kali_egress_to_teamserver_instances" {
```
Comment (dav3r) [inline kali_sg.tf]: Should we mention the ports in the resource name like we do elsewhere in this file?  I'm not tied to this suggestion, feel free to change it:
```suggestion
resource "aws_security_group_rule" "kali_egress_to_teamserver_instances_via_5000s" {
```

File: kali_sg.tf [hunk 4]
```diff
@@ -65,6 +51,30 @@ resource "aws_security_group_rule" "ingress_from_anywhere_to_kali_via_allowed_po
...
+  for_each = toset(["tcp", "udp"])
+
+  security_group_id        = aws_security_group.kali.id
+  type                     = "egress"
+  protocol                 = each.key
+  source_security_group_id = aws_security_group.teamserver.id
+  from_port                = 5000
+  to_port                  = 5999
+}
+resource "aws_security_group_rule" "kali_ingress_from_teamserver_instances" {
```
Comment (dav3r) [inline kali_sg.tf]: Same issue as above.
```suggestion
resource "aws_security_group_rule" "kali_ingress_from_teamserver_instances_via_5000s" {
```

File: teamserver_sg.tf [hunk 5]
```diff
@@ -81,3 +67,27 @@ resource "aws_security_group_rule" "ingress_from_anywhere_to_teamserver_via_allo
...
+resource "aws_security_group_rule" "teamserver_egress_to_kali_instances" {
```
Comment (dav3r) [inline teamserver_sg.tf]:
```suggestion
resource "aws_security_group_rule" "teamserver_egress_to_kali_instances_via_5000s" {
```

File: teamserver_sg.tf [hunk 6]
```diff
@@ -81,3 +67,27 @@ resource "aws_security_group_rule" "ingress_from_anywhere_to_teamserver_via_allo
...
+  for_each = toset(["tcp", "udp"])
+
+  security_group_id        = aws_security_group.teamserver.id
+  type                     = "egress"
+  protocol                 = each.key
+  source_security_group_id = aws_security_group.kali.id
+  from_port                = 5000
+  to_port                  = 5999
+}
+resource "aws_security_group_rule" "teamserver_ingress_from_kali_instances" {
```
Comment (dav3r) [inline teamserver_sg.tf]:
```suggestion
resource "aws_security_group_rule" "teamserver_ingress_from_kali_instances_via_5000s" {
```

**Review by dav3r — CHANGES_REQUESTED** [review]
I have a couple of suggested changes- let me know what you think.

File: teamserver_sg.tf (hunk 2 above)
Comment (jsf9k) [reply teamserver_sg.tf]: Please see commit 2fc7914.

File: kali_sg.tf (hunk 1 above)
Comment (jsf9k) [reply kali_sg.tf]: Please see commit 2fc7914.

File: teamserver_sg.tf (hunk 6 above)
Comment (jsf9k) [reply teamserver_sg.tf]: Please see commit b94e40a.

File: teamserver_sg.tf (hunk 5 above)
Comment (jsf9k) [reply teamserver_sg.tf]: Please see commit b94e40a.

File: kali_sg.tf (hunk 4 above)
Comment (jsf9k) [reply kali_sg.tf]: Please see commit b94e40a.

File: kali_sg.tf (hunk 3 above)
Comment (jsf9k) [reply kali_sg.tf]: Please see commit b94e40a.

**Comment by jsf9k** [comment]
@dav3r - Note that I rebased and force pushed in order to remove the deletion of the IMAP and Cobalt Strike security group rule; now that they are only requesting ports 5000-5999 (instead of 0-65535) that rule must return.

**Review by dav3r — APPROVED** [review]
Looks great! 🎸

**Review by mcdonnnj — APPROVED** [review]
:shipit: 🚀

**Pull request merged**